### ISSUE TITLE
Support user facing mechanism for applying third party Settings plugins

### ISSUE THREAD

**Issue by JLLeitschuh** [gradle/gradle#6710]
According to @oehme during a conversation I had on slack, supposedly you can also have `Settings` and `Gradle` plugins.

Currently, `Settings` for `settings.gradle.kts` doesn't allow a user to specify a `plugins` block inside of it.

There needs to be a mechanism added to support allowing users to specify 

### Expected Behavior
Should be able to specify something like this in a `settings.gradle.kts` file.
```kotlin
plugins {
    id ("my.plugin.name") version "1.2.34"
}
```

### Current Behavior
The above doesn't compile nor do I think it would work in Groovy either. 

### Context
I've found myself pasting this blob of code in all of my Gradle projects I've been creating both for work and for open source projects. I'd love to make it one Gradle plugin that I can just apply and be done with.

```kotlin
/**
 * This configures the gradle build so we can use non-standard build file names.
 * Additionally, this project can support sub-projects who's build file is written in Kotlin.
 *
 * @param project The project to configure.
 */
fun configureGradleBuild(project: ProjectDescriptor) {
    val projectBuildFileBaseName = project.name

    val gradleBuild = File(project.projectDir, "$projectBuildFileBaseName.gradle")
    val kotlinBuild = File(project.projectDir, "$projectBuildFileBaseName.gradle.kts")
    assert(!(gradleBuild.exists() && kotlinBuild.exists())) {
        "Project ${project.name} can not have both a ${gradleBuild.name} and a ${kotlinBuild.name} file. " +
                "Rename one so that the other can serve as the base for the project's build"
    }
    project.buildFileName = when {
        gradleBuild.exists() -> gradleBuild.name
        kotlinBuild.exists() -> kotlinBuild.name
        else ->
            throw AssertionError("Project `${project.name}` must have a either a file " +
                    "containing ${gradleBuild.name} or ${kotlinBuild.name}")
    }

    // Any nested children projects also get configured.
    project.children.forEach { configureGradleBuild(it) }
}

configureGradleBuild(rootProject)
```

### Your Environment
Gradle version `4.10`.

**Comment by JLLeitschuh** [comment]
Okay, upon further inspection, this is valid.
```kotlin
buildscript { 
    // plugin added to classpath here   
}
```

That being said, the `plugin` API is still missing.

Also, the fact that you can create plugins for `Settings` is, as far as I'm aware, completely undocumented.

**Comment by JLLeitschuh** [comment]
Use case: https://github.com/GradleWeaver/better-build-file-names

**Comment by JLLeitschuh** [comment]
This is now resolved in Gradle 6.0! Woot!

@ldaley @big-guy We should remove this bit of documentation:
https://docs.gradle.org/current/userguide/plugins.html#sec:build_scripts_only

**Comment by jjohannes** [comment]
This feature is available in Gradle 6+
